resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -502,6 +521,25 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
     )
   }
 
+  private def satisfiesLocality(offerHostname: String): Boolean = {
+    if (hostToLocalTaskCount.nonEmpty) {
```

Comment:
You've agreed that the semantics should be "Launch an executor on a host only when we have a task that wants to be on that host, or the configurable delay has elapsed"

By launching tasks on any arbitrary host when no locality info is available, you're violating those semantics, because even before the delay has elapsed, the scheduler will launch a task on an agent that no task wants to be on.  Does that make sense?